plugins/speculation-rules/settings.php

```diff
@@ -281,10 +309,74 @@ function plsr_render_settings_field( array $args ): void {
 		<?php endforeach; ?>
 
 		<p class="description" style="max-width: 800px;">
-			<?php echo esc_html( $args['description'] ); ?>
+			<?php
+			echo wp_kses(
+				$args['description'],
+				array(
+					'a' => array(
+						'href'   => array(),
+						'target' => array(),
+					),
+				)
+			);
+			?>
 		</p>
+
+		<?php if ( $show_notice ) : ?>
+			<div id="plsr-auth-notice" class="notice <?php echo $show_warning ? 'notice-warning' : 'notice-info'; ?> inline">
+				<p>
+					<strong id="plsr-notice-label" <?php echo $show_warning ? '' : 'hidden'; ?>><?php esc_html_e( 'Warning: ', 'speculation-rules' ); ?></strong>
+					<?php
+					echo wp_kses(
+						sprintf(
+							/* translators: %s: URL to persistent object cache documentation */
+							__( 'Enabling speculative loading for authenticated users without a persistent object cache may significantly increase server load. Consider setting up a <a href="%s" target="_blank">persistent object cache</a> before enabling this feature for logged-in users.', 'speculation-rules' ),
+							'https://developer.wordpress.org/advanced-administration/performance/optimization/#object-caching'
+						),
+						array(
+							'a' => array(
+								'href'   => array(),
+								'target' => array(),
+							),
+						)
+					);
+					?>
+				</p>
+			</div>
+		<?php endif; ?>
 	</fieldset>
+
 	<?php
+	if ( $show_notice ) {
+		wp_print_inline_script_tag(
+			sprintf(
+				'const authRadios = document.querySelectorAll( ".plsr-auth-radio" );
```

Comment:
```suggestion
				'const authRadios = document.querySelectorAll( "input[name='plsr_speculation_rules[authentication]']" );
```

Reply:
Will need to escape the single quotes:


```suggestion
				'const authRadios = document.querySelectorAll( "input[name=\'plsr_speculation_rules[authentication]\']" );
```